src/coq_rules.ml

```diff
@@ -112,14 +121,15 @@ let create_ccoq sctx ~dir =
   }
 
 (* get_libraries from Coq's ML dependencies *)
-let libs_of_coq_deps ~loc ~scope libs =
-  let lib_db = Scope.libs scope in
-  List.concat_map ~f:Dune_file.Lib_dep.to_lib_names libs
-  |> Lib.DB.find_many ~loc lib_db
-  |> Result.ok_exn
+let libs_of_coq_deps ~lib_db =
+  List.concat_map ~f:(fun lib_dep ->
+    let loc = Dune_file.Lib_dep.get_loc lib_dep in
+    Dune_file.Lib_dep.to_lib_names lib_dep
```

Comment:
I should have mentioned this earlier, but this isn't really handling `select` dependencies correctly. IMO, if such a feature isn't strictly necessary in coq then it shouldn't be supported. The way it's done for OCaml is annoyingly complicated, so if coq can do without it, that would be ideal. Otherwise, it should be properly supported as the current behavior is just confusing.

Reply:
Good point, I am indeed not sure we want to support `select` dependencies, I will switch the implementation to direct-style for now.

Reply:
Submitting a different PR with improvements here.